grails-app/controllers/org/zenboot/portal/processing/ScriptletBatchController.groovy

```
@@ -77,8 +77,10 @@ class ScriptletBatchController implements ApplicationEventPublisherAware{
             def execList = accessService.accessCache[springSecurityService.getCurrentUserId()].findAll { it.value == true}
             batches = new ArrayList<ScriptletBatch>()
             execList.each { key, value ->
-                Set<ExecutionZoneAction> actions = ExecutionZone.get(key).actions
-                actions.each {batches.addAll(it.scriptletBatches)}
+                if(key == params.filter?.executionZoneAction?.executionZone?.id?.toInteger() || params.filter == null) {
```

The params.filter = null check should probably be first and then the rest should be checked. That way it fails faster.